sw/device/sca/aes_serial.c

```diff
@@ -136,10 +120,16 @@ static void aes_key_config(const uint8_t *key, size_t key_len) {
 static void aes_key_mask_and_config(const uint8_t *key, size_t key_len) {
   SS_CHECK(key_len == kAesKeyLength);
   dif_aes_key_share_t key_shares;
-  prng_rand_bytes((uint8_t *)key_shares.share1, key_len);
+  // Mask the provided key.
   for (int i = 0; i < key_len / 4; ++i) {
+    key_shares.share1[i] = next_lfsr();
```

Comment:
Just chiming in from the PRNG side, `next_lfsr()` is not an ideal PRNG for masking keys because it has very low difusion. In this case, for example bits 0, 33, 66 and 99 will all be masked with the same bit value. This can cause leakage if these bits are somehow combined in the datapath.
Of course, if the key is re-masked in HW before using then it's OK.

Reply:
It might be helpful to run `next_lfsr()` for 32 steps instead of just one, e.g. [here](https://github.com/vrozic/opentitan/tree/next_lfsr) .

Reply:
Thanks @vrozic , I've included your commit into this PR (hope this is okay for you).